functions/src/index.ts

```diff
@@ -0,0 +1,188 @@
+import admin = require("firebase-admin");
+import XLSX = require("xlsx");
+import functions = require("firebase-functions");
+// const serviceAccount = require("../firebase.json");
+
+admin.initializeApp();
+
+/**
+ * Takes nested JSON object and flattens to one level
+ * @param ob original multi-level JSON object
+ * @param prefix prefix to attach to keys
+ * @returns sortedJSON - JSON sorted by keys alphabetically
+ */
+function flattenObject(ob: any, prefix: any): { [key: string]: any[] } {
+  const toReturn: { [key: string]: any[] } = {};
+  const newPrefix = prefix ? `${prefix}.` : "";
+
+  Object.keys(ob).forEach(function(i) {
+    if (typeof ob[i] === "object" && ob[i] !== null) {
+      // Recursion on deeper objects
+      Object.assign(toReturn, flattenObject(ob[i], newPrefix + i));
+    } else if (ob[i] === true) {
+      toReturn[prefix + i] = "Yes" as any;
+    } else if (ob[i] === false) {
+      toReturn[prefix + i] = "No" as any;
+    } else {
+      toReturn[prefix + i] = ob[i];
+    }
+  });
+  const sortedJSON = Object.keys(toReturn)
+    .sort()
+    .reduce((acc, key) => ({ ...acc, [key]: toReturn[key] }), {});
+  return sortedJSON;
+}
+
+/**
+ * Creates array of start and end indices of cells to be nested
+ * @param indices the start indices of each subheader
+ * @param totalLength total length of the header row
+ * @returns merged - array of cells to merge in excel document
+ */
+function mergedCells(indices: number[], totalLength: number): any {
+  const merged: any = [];
+  for (let i = 0; i < indices.length - 1; i += 1) {
+    merged.push({
+      s: { r: 0, c: indices[i] },
+      e: { r: 0, c: indices[i + 1] - 1 },
+    });
+  }
+  merged.push({
+    s: { r: 0, c: indices[indices.length - 1] },
+    e: { r: 0, c: totalLength - 1 },
+  });
+  return merged;
+}
+
+/**
+ * Creates main and sub headers and returns headers along with cells to be merged
+ * @param mainHeaders the names of the main header fields
+ * @param headers the names of all the subheaders
+ * @returns The main headers with spaces and subheaders, along with the cells to be merged
+ */
+function sortHeaders(mainHeaders: string[], headers: string[]): any {
+  const result: any = [];
+  const indices: any = [];
+  headers.forEach(function(header) {
```

Comment:
I would recommend being consistent with either using `function(...) { ... }` or arrow functions `(...) => { ... }` throughout. When passing in functions, I have a preference for arrow functions because they new/fancy, but also some actual reasons:

- https://babeljs.io/docs/en/learn#arrows-and-lexical-this
- https://stackoverflow.com/questions/34361379/are-arrow-functions-and-functions-equivalent-exchangeable

Regardless, I think consistency is important, so I recommend picking one and sticking with it.

Example of what I mean:
```typescript
array.forEach((...args) => { ... /* do something with args */ };
```
vs.
```typescript
array.forEach(function(args) { ... /* do something with args */ };
```